utility/src/test/java/org/oppia/android/util/parser/html/PolicyPageTagHandlerTest.kt

```diff
@@ -0,0 +1,187 @@
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.Captor
+import org.mockito.Mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.junit.MockitoJUnit
+import org.mockito.junit.MockitoRule
+import org.oppia.android.app.model.PolicyPage
+import org.oppia.android.testing.mockito.capture
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.ConsoleLogger
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.parser.html.CustomHtmlContentHandler.CustomTagHandler
+import org.oppia.android.util.parser.html.PolicyPageTagHandler.PolicyPageLinkClickListener
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+
+private const val POLICY_PAGE_LINK_MARKUP_1 =
+  "By using %s, you agree to our <br> <oppia-noninteractive-policy link=\"tos\"> Terms of " +
+    "Service </oppia-noninteractive-policy> and <oppia-noninteractive-policy link=\"privacy\">" +
+    "Privacy Policy </oppia-noninteractive-policy>."
+
+/** Tests for [PolicyPageTagHandler]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class PolicyPageTagHandlerTest {
+  @Rule
+  @JvmField
+  val mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+  @Mock
+  lateinit var mockImageRetriever: FakeImageRetriever
+
+  @Mock
+  lateinit var mockPolicyPageLinkClickListener: PolicyPageLinkClickListener
+
+  @Captor
+  lateinit var policyPageCaptor: ArgumentCaptor<PolicyPage>
+
+  @Inject
+  lateinit var context: Context
+  @Inject lateinit var consoleLogger: ConsoleLogger
+
+  private lateinit var noTagHandlers: Map<String, CustomTagHandler>
+  private lateinit var tagHandlersWithPolicyPageSupport: Map<String, CustomTagHandler>
+  private lateinit var testView: TextView
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    noTagHandlers = mapOf()
+    tagHandlersWithPolicyPageSupport = mapOf(
+      CUSTOM_POLICY_PAGE_TAG to PolicyPageTagHandler(
+        mockPolicyPageLinkClickListener,
+        consoleLogger
+      )
+    )
+    testView = TextView(context)
+  }
+
+  // TODO(#3085): Introduce test for verifying that the error log scenario is logged correctly.
+
+  @Test
+  fun testParseHtml_emptyString_doesNotIncludeClickableSpan() {
+    val parsedHtml =
+      CustomHtmlContentHandler.fromHtml(
+        html = "",
+        imageRetriever = mockImageRetriever,
+        customTagHandlers = tagHandlersWithPolicyPageSupport
+      )
+
+    val clickableSpans = parsedHtml.getSpansFromWholeString(ClickableSpan::class)
+    assertThat(clickableSpans).isEmpty()
+  }
+
+  @Test
+  fun testParseHtml_withPolicyPageMarkup_includesClickableSpan() {
```

Comment:
This should probably be split into two tests. Maybe have 2 HTML constants for each of the privacy policy & TOS cases.

Reply:
Or I can simply used span index to add clicks. Does it make sense? I have updated it. PTAL.